### ISSUE TITLE
W10/MSYS2 build breaks with last lensfun dev version

### ISSUE THREAD

**Issue by agriggio** [artpixls/ART#311]
**[Original report](https://bitbucket.org/agriggio/art/issues/311) by Gaaned92 (Bitbucket: [Gaaned92](https://bitbucket.org/Gaaned92), GitHub: [Gaaned92](https://github.com/Gaaned92)).**

----------------------------------------

Each time a new stable ART version is tagged, I update Lensfun to the last commit of master branch.

Doing that to build ART 1.21.1,  I checked out the lensfun version  master\_v0.3.95-1378-gde21e9fc.

Then the compilation of ART  stops with following error :

```none
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc: In destructor 'virtual rtengine::LFModifier::~LFModifier()':
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc:35:16: error: 'struct lfModifier' has no member named 'Destroy'
   35 |         data_->Destroy();
      |                ^~~~~~~
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc: In static member function 'static bool rtengine::LFDatabase::init(const Glib::ustring&)':
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc:320:35: error: 'Create' is not a member of 'lfDatabase'
  320 |     instance_.data_ = lfDatabase::Create();
      |                                   ^~~~~~
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc: In destructor 'rtengine::LFDatabase::~LFDatabase()':
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc:394:16: error: 'struct lfDatabase' has no member named 'Destroy'
  394 |         data_->Destroy();
      |                ^~~~~~~
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc: In member function 'std::unique_ptrrtengine::LFModifier rtengine::LFDatabase::getModifier(const rtengine::LFCamera&, const rtengi
ne::LFLens&, float, float, float, int, int, bool) const':
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc:496:43: error: 'Create' is not a member of 'lfModifier'
  496 |             lfModifier *mod = lfModifier::Create(lens.data_, camera.getCropFactor(), width, height);
      |                                           ^~~~~~
D:/ART/ARTSOURCE/ART/rtengine/rtlensfun.cc:501:26: error: 'struct lfModifier' has no member named 'Initialize'
  501 |             flags = mod->Initialize(lens.data_, LF_PF_F32, focalLen, aperture, focusDist > 0 ? focusDist : 1000, 0.0, LF_RECTILINEAR, flags, false);
      |                          ^~~~~~~~~~
make[2]: *** [rtengine/CMakeFiles/rtengine.dir/build.make:1280: rtengine/CMakeFiles/rtengine.dir/rtlensfun.cc.obj] Error 1
```

‌

If I come back to lensfun\_master\_v0.3.95-1344-g03e4a105, compilation of ART is OK.

It seems that in Revision: d67a1f45599dedd6ce686097ddcccc4f3ab83a4b all deprecated entries were removed.

>  Revision: d67a1f45599dedd6ce686097ddcccc4f3ab83a4b  
> Author: Joachim Daehnke [tuxfanx@gmail.com](mailto:tuxfanx@gmail.com)  
> Date: 26/12/2023 15:23:12  
> Message:  
> Merge pull request #2134 from payano/compile\_warnings
>
> Remove some compile warnings, remove deprecated functions
>
> Modified: CMakeLists.txt  
> Modified: include/lensfun/lensfun.h.in  
> Modified: libs/lensfun/database.cpp  
> Modified: libs/lensfun/lens.cpp  
> Modified: libs/lensfun/mod-coord.cpp  
> Modified: libs/lensfun/modifier.cpp  
> Modified: tests/deprecated/test\_database\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_color\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_coord\_centering\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_coord\_distortion\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_coord\_geometry\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_coord\_perspective\_correction\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_coord\_scale\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_coord\_tiny\_image\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_regression\_old.cpp  
> Modified: tests/deprecated/test\_modifier\_subpix\_old.cpp  
> Modified: tests/test\_database.cpp  
> Modified: tests/test\_lffuzzystrcmp.cpp  
> Modified: tests/test\_modifier.cpp  
> Modified: tests/test\_modifier\_coord\_centering.cpp  
> Modified: tests/test\_modifier\_coord\_distortion.cpp  
> Modified: tests/test\_modifier\_coord\_geometry.cpp  
> Modified: tests/test\_modifier\_coord\_perspective\_correction.cpp  
> Modified: tests/test\_modifier\_coord\_scale.cpp  
> Modified: tests/test\_modifier\_coord\_tiny\_image.cpp  
> Modified: tests/test\_modifier\_performance.cpp  
> Modified: tests/test\_modifier\_regression.cpp  
> Modified: tests/test\_modifier\_subpix.cpp

Alberto, do you intend to update ART accordingly? otherwise, I fear I will be stuck with a previous version.

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

Thanks for the info. Yes, I'll take care of it (when time permits)

**Comment by agriggio** [comment]
**Original comment by agriggio (Bitbucket: [agriggio](https://bitbucket.org/agriggio), GitHub: [agriggio](https://github.com/agriggio)).**

----------------------------------------

fixed compilation errors with lensfun master

Fixes #311
